hearpreprocess/runner.py

```diff
@@ -104,16 +104,21 @@ def run(
     else:
         sample_rates = [sample_rate]
 
-    tasks_to_run = [
-        task_script.main(  # type: ignore
+    tasks_to_run = []
+    for task_module in tasks[task]:
+        task_config = copy.deepcopy(task_module.generic_task_config)
+        if small:
+            task_config.update(dict(task_config["small"]))  # type: ignore
+        task_config.update({"tmp_dir": tmp_dir})
+        metadata_task = task_module.extract_metadata_task(task_config)
+        final_task = pipeline.FinalizeCorpus(
             sample_rates=sample_rates,
-            tmp_dir=tmp_dir,
             tasks_dir=tasks_dir,
             tar_dir=tar_dir,
-            small=small,
+            metadata_task=metadata_task,
+            task_config=task_config,
```

Comment:
Are these changes reflected in FinalizeCorpus? tmp_dir is passed in metadata now?

Reply:
Actually tmp_dir was never used but I fix it in #62 (see the luigi.py there)